src/app/beer_garden/requests.py

```diff
@@ -602,6 +610,9 @@ def process_request(
         # An error publishing means this request will never complete, so remove it
         db.delete(request)
 
+        if wait_timeout != 0:
+            request_map.pop(request.id, None)
+
```

Comment:
At this point, if we get an exception, no one is currently in a "wait" status. That occurs are line 629. So removing the item from the request_map is purely a clean up function so this list doesn't grow over long periods of time. I only did the wait_timeout because I was mirroring the function on line 595.

Reply:
Ahh I see. :+1: